### PR TITLE
fix: edge case namespace merkle tree

### PR DESCRIPTION
**Pull request by rach-id**
## Overview

This fixes an edge case with the `verifyInner` for the NMT implementation. It is the fix for `OS-CLST-ADV-01` from the Ottersec audit.

Special thanks for Ottersec on providing the test cases and the fix.

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

## Summary by CodeRabbit

- **Refactor**
  - Improved the calculation logic for nodes in the Merkle Tree structure.
  
- **Tests**
  - Added new tests to ensure the integrity of leaf and internal nodes in the Merkle Tree.

### CODE REVIEW

File: src/lib/tree/namespace/test/NamespaceMerkleTree.t.sol [hunk 1]
```diff
@@ -252,6 +252,271 @@ contract NamespaceMerkleTreeTest is DSTest {
         assertTrue(isValid);
     }
 
+    function testVerifyLeafFiveOfEight() external {
```
Comment (evan-forbes) [inline src/lib/tree/namespace/test/NamespaceMerkleTree.t.sol:255]: as you said we should figure out these tests are needed and document why they are needed so when they fail in the future we can better fix things

so we should create an issue imo to do so later, as I'm not sure we want to block on it due to the time constraints

**Review by evan-forbes — APPROVED** [review]

**Review by adlerjohn — APPROVED** [review]
